server/lib/uploader.js

```diff
@@ -35,17 +35,24 @@ const { uuid } = require('./util');
 const BadRequest = require('./errors/BadRequest');
 
 // configure the uploads directory based on global process variables
-const defaultDir = 'uploads'; // NOTE: this must be a relative path
-const dir = process.env.UPLOAD_DIR || defaultDir; // relative path
-const fsdir = path.join(process.cwd(), dir); // global path
+const defaultDir = 'uploads';
+const dir = process.env.UPLOAD_DIR || defaultDir;
+// NOTE: 'dir' must be a relative path (for http requests to work)
+if (path.isAbsolute(dir) || dir.startsWith('..')) {
+  throw new Error(`UPLOAD_DIR (${dir}) must be a relative path within the BHIMA software installation!`);
+}
+const rootDir = path.normalize(`${process.cwd()}/..`);
```

Comment:
This isn't guaranteed to be the root directory in all circumstances.  As explained in [this SO answer](https://stackoverflow.com/a/9874415), the `process.cwd()` is wherever the `node` call was made.  So, it can changed, depending on where the executing terminal is in the file system.

Instead, you probably want `__dirname`.   [Here is a really good tutorial explaining how it works](https://www.digitalocean.com/community/tutorials/nodejs-how-to-use__dirname).  This will never change, no matter who calls it or from where it is called.